domain/machine/state/machine_cloud_instance.go

```diff
@@ -148,3 +150,52 @@ WHERE machine_uuid=$instanceTag.machine_uuid
 		return nil
 	})
 }
+
+// InstanceId returns the cloud specific instance id for this machine.
+// If the machine is not provisioned, it returns a NotProvisionedError.
+func (st *State) InstanceId(ctx context.Context, machineId coremachine.ID) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	machineIDParam := sqlair.M{"machine_id": machineId}
+	query := `
+SELECT instance_id AS &instanceID.*
+FROM machine AS m
+         JOIN machine_cloud_instance AS mci ON m.uuid = mci.machine_uuid
+WHERE m.machine_id = $M.machine_id;
+`
+	queryStmt, err := st.Prepare(query, sqlair.M{}, instanceID{})
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	var instanceId string
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		result := instanceID{}
+		err := tx.Query(ctx, queryStmt, machineIDParam).Get(&result)
+		if err != nil {
+			return errors.Annotatef(err, "querying instance id for machine %q", machineId)
+		}
+
+		instanceId = result.ID
+		return nil
+	})
+	if err != nil {
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return "", errors.Annotatef(machineerrors.NotProvisioned, "machine id: %q", machineId)
+		}
+		return "", errors.Trace(err)
+	}
+	return instanceId, nil
+}
+
+// InstanceStatus returns the cloud specific instance status for this
+// machine.
+// If the machine is not provisioned, it returns a NotProvisionedError.
+func (st *State) InstanceStatus(ctx context.Context, machineId coremachine.ID) (string, error) {
+	// TODO(cderici): Implementation for this is deferred until the design for
+	// the domain entity statuses on dqlite is finalized.
+	return "", errors.NotImplemented
```

Comment:
Just return a real status and keep the TODO. This will throw errors in integration tests.